coralproject/talk
# [CORL-2934]: DSA - updates to design and copy

## What does this PR do?

These changes include various updates to design and copy related to reporting and moderating illegal content for DSA. They include:
* updates to the validation for adding additional comments for reporting illegal content. You will get an error upon clicking `Add comment URL` if your comment URL: 1) is not a valid URL, 2) is for a comment already being reported in this DSA report 3) does not contain a valid comment id that exists 4) contains the comment id for a comment that the user has already previously reported for illegal content. In each case, the correct error should be displayed.
* style updates to the individual illegal content report page in the admin, including 1) decision section styles updated 2) status icons and dropdown to change status updated
* style updates to the illegal content reports table in the admin, including 1) status icons updated
* add ability to remove additional comments after they are added

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

To test validation updates for adding multiple additional comments to a report
You can test this PR by clicking on `Report` for a comment in the stream and then going to its illegal content report form. Add additional comments to the report form. Test that if valid URL for a valid comment, they are added as expected. If you try to add a comment twice, you see the appropriate error. If you try to add an invalid URL, you see the appropriate error. If you add a valid URL with an invalid comment id (comment doesn't exist for that id), then you see the appropriate error. If you add a valid URL for a valid comment id but the user has already previously reported that comment for illegal content, then you see the appropriate error.

* add additional comments to a report form and then delete them from it

Design updates to look at
* Decision section of a completed illegal content report
* Status icons in overall reports table in the admin, as well as status icons and dropdown on individual reports pages

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
